### PR TITLE
feat(#26598): add --java-version so init don't enter into unwanted interactive mode

### LINKED ISSUE #26598: add --java-version parameter to gradle init task
### Expected Behavior

```bash
gradle init \
  --type java-application \
  --dsl groovy \
  --test-framework junit-jupiter \
  --package sample.structure \
  --project-name my-gradle-project  \
  --no-split-project  \
  --java-version 17
```

### Current Behavior (optional)

_No response_

### Context

the proposed command line presents us an error since the `--java-version` parameter does not exists.

when removed it ends up falling into the interactive mode where we can provide a value for java version.

my need (and maybe others as well) is to provision the new project without any interactive mode.

### PR DESCRIPTION
**Pull request by sombriks**
Fixes #26598 

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by tresat — CHANGES_REQUESTED** [review]
Thank you for the submission and your interest in contributing to Gradle!  

Unfortunately we can't accept this as-is.  We do appreciate the effort that went into this work, and I think with a few further revisions this change could be added.

I've left some feedback and suggestions.  Due to the nature of the change, other Gradle maintainers will likely also take a look before this is merged.

**Comment by tresat** [comment]
❌ Also be sure to run `gradlew sanityCheck` and fix any reported issues there.  New public API properties must be marked `@Incubating`.

File: subprojects/docs/src/docs/userguide/core-plugins/build_init_plugin.adoc
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/core-plugins/build_init_plugin.adoc:87]:
```suggestion
You can see available options using the `help` task:
```

File: subprojects/docs/src/docs/userguide/core-plugins/build_init_plugin.adoc
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/core-plugins/build_init_plugin.adoc:93]:
```suggestion
The build type can be specified by using the `--type` command-line option. 
For example, to create a Java library project run:
```

**Review by lkasso — APPROVED** [review]
Some minor docs to cleanup.

**Comment by sombriks** [comment]
Hi @tresat , @lkasso , thanks for the review!
I'll address the comments carefully and then ask for another round!

File: platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java
Comment (tresat) [inline platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java:138]:
```suggestion
     * @return the java version number supplied by the user
```

File: platforms/software/build-init/src/testFixtures/groovy/org/gradle/buildinit/plugins/fixtures/ScriptDslFixture.groovy
Comment (tresat) [inline platforms/software/build-init/src/testFixtures/groovy/org/gradle/buildinit/plugins/fixtures/ScriptDslFixture.groovy:83]:
```suggestion
    void assertGradleFilesGenerated(TestFile parentFolder = rootDir, String... pathForBuild) {
```

**Review by tresat — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**
